### PR TITLE
DOC: clarify default window in rolling method

### LINKED ISSUE #17893: Not clear which window is default for df.rolling()
The docstring for `df.rolling()` doesn't say which window is the default (I assume it's boxcar?): https://pandas-docs.github.io/pandas-docs-travis/generated/pandas.DataFrame.rolling.html

### PR DESCRIPTION
**Pull request by dstansby**
- [x] closes #17893 
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/window.py [hunk 1]
```diff
@@ -503,6 +503,7 @@ class Window(_Window):
     * ``general_gaussian`` (needs power, width)
     * ``slepian`` (needs width).
 
+    If ``win_type=None`` a boxcar window is used.
```
Comment (jorisvandenbossche) [inline pandas/core/window.py]: If we add this, I think we should explain what a boxcar window is (as now in the default case the user does not need to worry about it, and those assuming you just get a uniform hard-cut window are correct)

Comment (dstansby) [reply pandas/core/window.py]: Come to think of it, none of the other windows are explained either! Are the windows used custom pandas ones, or are they just the normal numpy ones? And if they're pandas ones do they have their own bit of documentation to link to?

Comment (jreback) [reply pandas/core/window.py]: these are all scipy windows
certainly take links to that documentation (in doc string and in computation.rst)

**Comment by dstansby** [comment]
I've added a link to the scipy docs - I presumed this is where all the windows come from?

File: pandas/core/window.py [hunk 2]
```diff
@@ -503,6 +503,9 @@ class Window(_Window):
     * ``general_gaussian`` (needs power, width)
     * ``slepian`` (needs width).
 
+    If ``win_type=None`` a boxcar window is used. To learn more about different
```
Comment (jreback) [inline pandas/core/window.py]: this is not correct, the default is in fact ``win_type=None``, which means your window is specified by the the ``window`` parameters; *only* if you actually specify a ``win_type`` do you get a specific weighted window.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window.py (hunk 2 above)
Comment (dstansby) [reply pandas/core/window.py]: Fundamentally doing a rolling window calculation involves choosing a window, so what window does passing `win_type=None` choose? I thought it was a boxcar?

Is it valid to specify both the window parameters and the window type, or can only one be provided?

Comment (jreback) [reply pandas/core/window.py]: most operations are an u weighted window
only if you specify win_type do you get a special weighted window (which also has only sum/mean) methods (this is not very common usage, not using win_type is by far the most common)

i think the docs in computation are pretty clear in this

certainly could add more / better examples in the doc string

Comment (dstansby) [reply pandas/core/window.py]: Right, so if you don't specify win_type you get an un-weighted window, which is a boxcar window, which is what my addition says.

I disagree that the docs are pretty clear, and I think this helps clear them up. (I also think it doesn't confuse anything, and I'm 99% sure what I've written is correct!)

Comment (jorisvandenbossche) [reply pandas/core/window.py]: Yes, the current default uniform unweighted window is apparently a boxcar window (which I only learned by looking up what a boxcar window is). 
I think that I am not alone in that, therefore I would change the explanation slightly to first explain what the default window is, and then only to say that this is called a boxcar window (people should certainly not need to follow a link to know what the meaning of the default is IMO)

Comment (dstansby) [reply pandas/core/window.py]: Ah, I think I understand the confusion now - I don't have time now, but will try and revise later.

**Comment by dstansby** [comment]
Is this a bit better?

**Comment by jreback** [comment]
seems ok to me, @jorisvandenbossche

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@jorisvandenbossche

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@dstansby Thanks a lot!

**Pull request merged**